### PR TITLE
Better commenting for addl params

### PR DESCRIPTION
**Pull request by dsgouda**
Addresses https://github.com/Azure/autorest/issues/1355

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/SwaggerModeler.cs [hunk 1]
```diff
@@ -293,20 +293,13 @@ public static string GetMethodName(Operation operation)
...
-                return operation.OperationId;
-            }
-
-            var parts = operation.OperationId.Split('_');
-            return parts[1];
+            return GetMethodNameFromOperationId(operation.OperationId);
         }
 
+        public static string GetMethodNameFromOperationId(string operationId) => 
+            (operationId?.IndexOf('_') != -1) ? operationId.Split('_')[1]: operationId;
```
Comment (olydis) [inline src/modeler/AutoRest.Swagger/SwaggerModeler.cs]: touching that code anyways, I'd just define the method as `operationId.Split('_').Last()`

File: src/core/AutoRest.Extensions/ParameterGroupExtensionHelper.cs [hunk 2]
```diff
@@ -111,8 +111,7 @@ private static string GenerateParameterGroupModelText(IEnumerable<Method> method
             if (methodList.Count == 1)
             {
                 Method method = methodList.Single();
-                return string.Format(CultureInfo.InvariantCulture, "Additional parameters for the {0} operation.",
-                    createOperationDisplayString(method.MethodGroup.Name.ToPascalCase(), method.Name));
+                return string.Format(CultureInfo.InvariantCulture, "Additional parameters for {0} operation.", method.Name);
```
Comment (olydis) [inline src/core/AutoRest.Extensions/ParameterGroupExtensionHelper.cs]: For consistency, shouldn't this use `SwaggerModeler.GetMethodNameFromOperationId(method.Name)` as above?
I'd just reuse `"Additional parameters for " + SwaggerModeler.GetMethodNameFromOperationId(method.Name) + " operation."` here, `string.Format` with `CultureInfo` is complete nonsense here anyways.

File: src/modeler/AutoRest.Swagger/SwaggerModeler.cs (hunk 1 above)
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/SwaggerModeler.cs]: Do we always assume that all operationIds MUST have a single underscore? I know we have a validation rule to enforce that but still seems like a sensitive change.

Comment (olydis) [inline src/modeler/AutoRest.Swagger/SwaggerModeler.cs]: Note that for multiple underscores, neither picking the 2nd nor the last one makes sense. So yes, we do assume it and making this simplification will at least be crystal clear with that (the other one only looks smarter when it really isn't 😆 )

File: src/core/AutoRest.Extensions/ParameterGroupExtensionHelper.cs [hunk 3]
```diff
@@ -110,9 +110,7 @@ private static string GenerateParameterGroupModelText(IEnumerable<Method> method
             List<Method> methodList = methodsWhichUseGroup.ToList();
             if (methodList.Count == 1)
             {
-                Method method = methodList.Single();
-                return string.Format(CultureInfo.InvariantCulture, "Additional parameters for the {0} operation.",
-                    createOperationDisplayString(method.MethodGroup.Name.ToPascalCase(), method.Name));
+                return "Additional parameters for "+ methodList.Single().Name + " operation.";
```
Comment (olydis) [inline src/core/AutoRest.Extensions/ParameterGroupExtensionHelper.cs]: still wondering, why is this inconsistent with the `Documentation` string you assemble above?

Comment (dsgouda) [inline src/core/AutoRest.Extensions/ParameterGroupExtensionHelper.cs]: If by above you mean `AzureExtensions.cs`, for some reason looks like the `method` variable there has `name` in the form of operationId itself while in `ParameterGroupExtensionHelper.cs`, the `name` property denotes the `{1}` in `{0}_{1}` in an operationId.
We could probably investigate where these diverge but looks like overkill for such a small change.

Comment (olydis) [inline src/core/AutoRest.Extensions/ParameterGroupExtensionHelper.cs]: OOOOHHH okay yeah typical AutoRest, no worries then, sorry about the nagging 👍

File: Samples/3f-example-generation/shell/stderr.txt [hunk 4]
```diff
@@ -1,6 +1,26 @@
+  at ...
+  at ...
```
Comment (olydis) [inline Samples/3f-example-generation/shell/stderr.txt]: @dsgouda uh I think there were some errors during generation, always look at the diff ;)

Comment (olydis) [inline Samples/3f-example-generation/shell/stderr.txt]: only regen the samples to fix this one...

Comment (dsgouda) [inline Samples/3f-example-generation/shell/stderr.txt]: On it.

**Pull request merged**